functions/concepts/retry-timeout/src/test/java/functions/RetryTimeoutTest.java

```diff
@@ -65,41 +64,29 @@ public void afterTest() {
 
   @Test
   public void retryTimeout_handlesRetryMsg() {
-    String timestampData = gson.toJson(Map.of(
-        "timestamp", ZonedDateTime.now(ZoneOffset.UTC).toString()));
+    ZonedDateTime timestamp = ZonedDateTime.now(ZoneOffset.UTC);
+    Context mockContext = mock(Context.class);
+    when(mockContext.timestamp()).thenReturn(timestamp.toString());
 
     PubSubMessage pubsubMessage = new PubSubMessage();
-    pubsubMessage.setData(timestampData);
 
-    new RetryTimeout().accept(pubsubMessage, null);
+    new RetryTimeout().accept(pubsubMessage, mockContext);
 
     String logMessage = LOG_HANDLER.getStoredLogRecords().get(0).getMessage();
-    assertThat(String.format("Processing event %s.", timestampData)).isEqualTo(logMessage);
+    assertThat(logMessage).contains("Processing event with timestamp " + timestamp);
   }
 
   @Test
   public void retryTimeout_handlesStopMsg() {
```

Comment:
Should this test contain an assertion? (I don't think it does right now.)

Reply:
If you're looking at the diffs it can be confusing, since I deleted the empty-payload test. The two remaining test methods are basically identical apart from the timestamp, and the assertion is on the last line of each.